packages/data-context/src/sources/RelevantRunSpecsDataSource.ts

```diff
@@ -208,7 +210,9 @@ export class RelevantRunSpecsDataSource {
           debug('Run statuses changed')
           const projectSlug = await this.ctx.project.projectId()
 
-          if (projectSlug) {
+          const wasWatchingCurrentProject = this.#cached.statuses.current === 'RUNNING'
+
+          if (projectSlug && wasWatchingCurrentProject) {
```

Comment:
This is the change that avoids a flash of the full page loading skeleton when a `next` run transitions from "running" to "completed".

Ideally we would have an end-to-end test for this, I'm working on one that can form part of a separate PR if this merges first. It's also the kind of test that will be much easier to write if we have improvements coming in the test infra soon.

Reply:
I don't think the `wasWatchingCurrentProject` check will work at its current location. The `#cached` variable is getting set on line 201, and the `statusesChanged` variable is calculated before that.  When the code gets to the `wasWatchingCurrentProject` calculation, the `#cached.statues.current` will no longer be RUNNING.

Reply:
Good point. It's how the code was in the video I put in the PR description where it's having the intended effect, let me take a look at why exactly that's the case.

Reply:
Updated to move this earlier in the flow, everything seems good anyway.